frontend/public/components/utils/headings.tsx

```diff
@@ -102,6 +105,18 @@ export const PageHeading = connectToModel((props: PageHeadingProps) => {
     (hasButtonActions || hasMenuActions) && hasData && !_.get(data, 'metadata.deletionTimestamp');
   const resourceStatus = hasData && getResourceStatus ? getResourceStatus(data) : null;
   const showHeading = props.icon || kind || resourceTitle || resourceStatus || badge || showActions;
+  const actionExtensions = useExtensions<KebabActionFactory>(isKebabActionFactory);
+
+  let menuOptions = _.isFunction(menuActions)
+    ? menuActions(kindObj, data, extraResources, customData)
+    : ((menuActions as KebabAction[]) || []).map((a) =>
+        a(kindObj, data, extraResources, customData),
+      );
+
+  if (extendMenuActions) {
+    menuOptions = mergePluginKebabOptions(menuOptions, actionExtensions, kindObj, data);
+  }
+
```

Comment:
we should compute `hasMenuActions` variable here and update dependent variables accordingly

Reply:
@suomiy Can you please elaborate on that idea?

Reply:
`hasMenuActions` may be false because the non-extended resource didnt have any specified - event after adding menu actions via extensions, they wouldnt show up because `hasMenuActions` is false https://github.com/openshift/console/pull/4721/files#diff-45aa3dfc05205bd3d1b3da31bd999e6dR169